src/main/java/com/revature/controllers/TripController.java

```diff
@@ -0,0 +1,140 @@
+     * @return A list of all the trips.
+     */
+    @ApiOperation(value = "Return all trips", tags = {"Trip"})
+    @GetMapping
+    public ResponseEntity<List<Trip>> getTrips() {
+        List<Trip> trips = tripService.getTrips();
+
+        if (trips.size() != 0) return ResponseEntity.ok(trips);
+
+        return ResponseEntity.noContent().build();
+    }
+
+    /**
+     * HTTP GET method (/trips/{number})
+     *
+     * @param id represent the trip's ID.
+     * @return A trip that matches the ID.
+     */
+    @ApiOperation(value = "Return trip by ID", tags = {"Trip"})
+    @GetMapping("/{id}")
+    public ResponseEntity<Trip> getTripById(@PathVariable("id") int id) {
+        Optional<Trip> trip = Optional.ofNullable(tripService.getTripById(id));
+
+        if (trip.isPresent()) return ResponseEntity.ok(trip.get());
+
+        return ResponseEntity.noContent().build();
+    }
+
+    /**
+     * HTTP GET method (/trips/driver/{driveId})
+     *
+     * @param driverId represents the driver's ID.
+     * @return A list of trips matching the driver's ID.
+     */
+    @ApiOperation(value = "Returns trips by driver ID", tags = {"Driver", "Trip"})
+    @GetMapping("/driver/{driverId}")
+    public ResponseEntity<List<Trip>> getTripsByDriverId(@PathVariable("driverId") int driverId) {
+        List<Trip> trips = tripService.getTripsByDriverId(driverId);
+
+        if (trips.size() != 0) return ResponseEntity.ok(trips);
+
+        return ResponseEntity.noContent().build();
+    }
+
+    /**
+     * HTTP GET method (/trips/rider/{riderId})
+     *
+     * @param riderId represents the rider's ID.
+     * @return A list of trips matching the rider's ID.
+     */
+    @ApiOperation(value = "Returns trips by rider ID", tags = {"Rider", "Trip"})
+    @GetMapping("/rider/{riderId}")
+    public ResponseEntity<List<Trip>> getTripsByRiderId(@PathVariable("riderId") int riderId) {
+        List<Trip> trips = tripService.getTripsByRiderId(riderId);
+
+        if (trips.size() != 0) return ResponseEntity.ok(trips);
+
+        return ResponseEntity.noContent().build();
+    }
+
+    /**
+     * HTTP POST method (/trips)
+     * @param trip represents the new Trip object being sent.
+     * @return The newly created object along with a 201 code.
+     */
+    @ApiOperation(value = "Adds a new trip")
+    @PostMapping
+    public ResponseEntity<Trip> addTrip(@Valid @RequestBody Trip trip) {
+        int oldId = trip.getTripId();
+        int newId = tripService.addTrip(trip).getTripId();
+
+        if (newId != oldId) return ResponseEntity.status(201).body(trip);
+
+        return ResponseEntity.badRequest().build();
+    }
+
+    /**
+     * HTTP PUT method (/trips)
+     *
+     * @param trip represent the updated Trip object being sent.
+     * @return The newly updated Trip object.
+     */
+    @ApiOperation(value = "Updates a trip by ID", tags = {"Trip"})
+    @PutMapping
+    public ResponseEntity<Trip> updateTrip(@Valid @RequestBody Trip trip) {
```

Comment:
I used this structure in previous demos, and it works alright, but it has a critical flaw: If you receive a bad request, you save it as a new entity. This is most likely not desired behavior.

Reply:
Thank you for pointing this out. Resolved.